src/host/getset.cpp

```diff
@@ -608,6 +608,18 @@ void ApiRoutines::GetLargestConsoleWindowSizeImpl(const SCREEN_INFORMATION& cont
             gci.SetLegacyColorTableEntry(i, data.ColorTable[i]);
         }
 
+        // GH#399: Trigger a redraw, so that updated colors are repainted, but
+        // only do this if we're not in conpty mode. ConPTY will update the
+        // colors of the palette elsewhere (after TODO GH#10639)
+        if (!gci.IsInVtIoMode())
+        {
+            auto* pRender = ServiceLocator::LocateGlobals().pRender;
+            if (pRender)
+            {
+                pRender->TriggerRedrawAll();

```

Comment:
this shouldn't introduce a poor performance case for applications that aren't intending on changing the colors, right?

Reply:
> inb4 some crazy app that's calling `SetConsoleScreenBufferInfoEx` in a tight loop

Reply:
I mean, it totally might. I guess I could check the colors to see if they actually changed.

Reply:
> > inb4 some crazy app that's calling `SetConsoleScreenBufferInfoEx` in a tight loop

That's not that crazy. Get/Set console screen buffer on each command is one of the few ways to bulk get/set that sort of state. I wouldn't be surprised if something like PSReadline was doing that without realizing.

Reply:
Every time this is called we `memset` 456 bytes, so it's not entirely cost free.
As far as I can see we don't have any engine anymore which don't already accumulate each row of text during rendering anyways, so we can technically drop any invalidation bitmaps and only use row-wise invalidations. That way the cost of `TriggerRedrawAll()` itself will be negligible. (There's still the cost of redrawing everything of course.)

Apart from that: Why don't we just check whether the palette actually changed here? Let's just cache it here and compare it?

Reply:
If you're worried about the performance of `TriggerRedrawAll`, it's worth noting that there's potentially another `TriggerRedrawAll` call a couple of lines down in the `SetDefaultAttributes` method (that only happens if one of the attributes has changed though).

Reply:
If it's not hard to check if they changed... we should do that before forcing a repaint. We've been bitten enough times by apps doing weird things with our APIs.